standard/lpdb/count.lua

```diff
@@ -0,0 +1,188 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Count
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+local Team = require('Module:Team')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local Count = {}
+
+
+function Count.games(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+
+	return Count._query('game', lpdbConditions)
+end
+
+function Count.matches(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('match', lpdbConditions)
+end
+
+function Count.tournaments(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args, true)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('tournament', lpdbConditions)
+end
+
+function Count.placements(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	if not Logic.readBool(args.includeShowmatch) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Showmatch')}
+	end
+
+	if not Logic.readBool(args.includeQualifier) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatier'), Comparator.neq, 'Qualifier')}
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Qualifier')}
+	end
+
+	if Logic.readBool(args.noEmptyPrizePool) then
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.neq, '')}
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.gt, 0)}
+	end
+
+	if String.isNotEmpty(args.player) then
+		local opponent = mw.ext.TeamLiquidIntegration.resolve_redirect(args.player)
+		local teamConditions = ConditionTree(BooleanOperator.any)
+		for index = 1, 10 do
+			teamConditions:add{
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponent),
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponent:gsub(' ', '_'))
```

Comment:
```suggestion
				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponentWithUnderscores)
```

- slightly better runtime since you gsub once instead of 10 times
- gsub returns 2 values (2nd one is the number of replacements) so this would mean passing a 4th param to ConditionNode which only eats 3 —> the extension in vscode would complain